python/cugraph/structure/graph.py

```diff
@@ -11,109 +11,32 @@
     if col.null_count != 0:
         raise ValueError("Series contains NULL values")
 
 
 class Graph:
-    class EdgeList:
-        def __init__(self, *args):
-            if len(args) == 1:
-                self.__from_dask_cudf(*args)
-            else:
-                self.__from_cudf(*args)
-
-        def __from_cudf(self, source, destination, edge_attr=None):
-            self.edgelist_df = cudf.DataFrame()
-            self.edgelist_df["src"] = source
-            self.edgelist_df["dst"] = destination
-            self.weights = False
-            if edge_attr is not None:
-                self.weights = True
-                if type(edge_attr) is dict:
-                    for k in edge_attr.keys():
-                        self.edgelist_df[k] = edge_attr[k]
-                else:
-                    self.edgelist_df["weights"] = edge_attr
-
-        def __from_dask_cudf(self, ddf):
-            self.edgelist_df = ddf
-            self.weights = False
-            # FIXME: Edge Attribute not handled
-
-    class AdjList:
-        def __init__(self, offsets, indices, value=None):
-            self.offsets = offsets
-            self.indices = indices
-            self.weights = value  # Should be a dataframe for multiple weights
-
-    class transposedAdjList:
-        def __init__(self, offsets, indices, value=None):
-            Graph.AdjList.__init__(self, offsets, indices, value)
-
-    """
-    cuGraph graph class containing basic graph creation and transformation
-    operations.
-    """
-
-    def __init__(
-        self,
-        m_graph=None,
-        symmetrized=False,
-        bipartite=False,
-        multi=False,
-        dynamic=False,
-    ):
-        """
-        Returns
-        -------
-        G : cuGraph.Graph.
-
-        Examples
-        --------
-        >>> import cuGraph
-        >>> G = cuGraph.Graph()
-
-        """
-        self.symmetrized = symmetrized
-        self.renumbered = False
-        self.renumber_map = None
-        self.bipartite = False
-        self.multipartite = False
-        self._nodes = {}
-        self.multi = multi
-        self.distributed = False
-        self.dynamic = dynamic
-        self.self_loop = False
-        self.edgelist = None
-        self.adjlist = None
-        self.transposedadjlist = None
-        self.edge_count = None
-        self.node_count = None
-
-        # MG - Batch
-        self.batch_enabled = False
-        self.batch_edgelists = None
-        self.batch_adjlists = None
-        self.batch_transposed_adjlists = None
+    class Properties:
+        def __init__(self, directed):
+            self.directed = directed
 
+    def __init__(self, m_graph=None, directed=False):
+        self.Impl = None
+        self.graph_properties = Graph.Properties(directed)
         if m_graph is not None:
-            if type(m_graph) is MultiGraph or type(m_graph) is MultiDiGraph:
+            if m_graph.is_multigraph():
                 elist = m_graph.view_edge_list()
```

Comment:
`view_edge_list()` no longer belongs to the graph class (now in class `simpleGraphImpl`). I am wondering how this line will work

Reply:
maybe `m_graph.Impl.view_edge_list()` ?

Reply:
I did not pay attention to `__getattr__`. That solves all my concerns